### DISCUSSION TITLE
通过vercel部署后，对话记录是保存在哪里的（服务器还是本地）？可否导出（备份）、导入？

### DISCUSSION THREAD

**Discussion by szdbin** [lobehub/lobehub#1168] [Self Hosting] [answered]
这个项目非常好，使用也很顺畅，但有些问题不太明白：
1、velcel部署后，每台设备访问都是相互独立的，是不是意味着所有的对话信息都保存在本地？
2、重新部署后，历史的对话信息还在吗？
3、每台设备的所有信息（对话信息、助手信息等）可否一键导出（备份）、导入？
谢谢！

**Comment by arvinxx** [comment, marked as answer]
>  1、velcel部署后，每台设备访问都是相互独立的，是不是意味着所有的对话信息都保存在本地？

是的，保存在你设备访问的本地浏览器 indexedDB 数据库中。

<img width="933" alt="image" src="https://github.com/lobehub/lobe-chat/assets/28616219/36d152aa-1bde-45c9-93f5-069fd0a76fa0">

> 2、重新部署后，历史的对话信息还在吗？

还在的，但是如果你换一个域名，就不在了。如果要切换域名的话，可以参考 #1127 

> 3、每台设备的所有信息（对话信息、助手信息等）可否一键导出（备份）、导入？

可以设置里导出：

<img width="406" alt="image" src="https://github.com/lobehub/lobe-chat/assets/28616219/e90d5369-12ac-4d39-9050-eab7b19342a9">

**Reply by szdbin** [reply]
感谢答疑！
